### PR TITLE
CMDCT-3977 + CMDCT-3978: NAAAR & Feature Flagging

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

This PR sets up the routes and bare-bones verbiage for NAAAR, which is all behind a LaunchDarkly feature flag.

### Related ticket(s)

CMDCT-3977
CMDCT-3978

---
### How to test

- Log in as a state user
- The home page should have a button to navigate to NAAAR:

Screenshot 2024-09-13 at 11 22 40 AM

- If you click on it or navigate directly to `/naaar`, you should see an empty dashboard page
- Log in as an admin user
- The admin dashboard should have a radio button option for the NAAAR report

To test with a "disabled" flag, search for `useFlags()?.naaarReport` and set these variables to `false` and you shouldn't be able to navigate to `/naaar` routes OR see the radio button in the admin dashboard ✨ 

Screenshot 2024-09-13 at 11 22 19 AM

### Notes

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/components/app/AppRoutes.tsx [hunk 1]
```diff
@@ -111,6 +108,45 @@ export const AppRoutes = () => {
...
+              )}
+              <Route
+                path="/naaar/*"
+                element={
+                  !contextIsLoaded ? (
+                    <Flex sx={sx.spinnerContainer}>
+                      <Spinner size="lg" />
+                    </Flex>
+                  ) : (
+                    <Navigate to="/mlr" />
```
Comment (gmrabian) [inline services/ui-src/src/components/app/AppRoutes.tsx:144]: probably want to redirect to /naaar, right?

File: services/ui-src/src/components/forms/AdminDashSelector.tsx [hunk 2]
```diff
@@ -30,11 +30,21 @@ export const AdminDashSelector = ({ verbiage }: Props) => {
...
   // assemble and inject report choices depending on whether report is enabled
   const mlrReport = useFlags()?.mlrReport;
+  const naaarReport = useFlags()?.naaarReport;
   const reportField = formJson.fields.find((field) => field.id === "report")!;
 
+  if (naaarReport) {
+    reportField.type = "radio";
+    reportField.validation = "radio";
+    reportChoices.push(naaarReportChoice);
+  }
```
Comment (gmrabian) [inline services/ui-src/src/components/forms/AdminDashSelector.tsx:47]: I noticed this button showed up between mcpar and mlr. Putting it at the end would keep things consistent alphabetically and with the state view

File: services/ui-src/src/utils/testing/mockZustand.tsx [hunk 3]
```diff
@@ -153,6 +154,18 @@ export const mockMlrLockedReportStore: McrReportState = {
   setLastSavedTime: () => {},
 };
 
+export const mockNaaarReportStore: McrReportState = {
+  report: mockNaaarReport,
+  reportsByState: [mockMcparReport],
```
Comment (gmrabian) [inline services/ui-src/src/utils/testing/mockZustand.tsx:159]: replace with mockNaaarReport, or leave empty

**Review by gmrabian — COMMENTED** [review]
Looks pretty good functionality-wise! A few things I'd like addressed before merging:

**Comment by karla-vm** [comment]
@gmrabian thanks for your feedback, great finds! I just addressed them in the latest push :)

**Review by jessabean — APPROVED** [review]

**Pull request merged**